### ISSUE TITLE
Evaluation inconsistency

### ISSUE THREAD

**Issue by iOiurson** [anthill/koala#19]
Evaluation does not always give the same result.

Example:
With `InputData!G14` as `2018` in the .XLS, 
````
print 'First evaluation', sp.evaluate('Cashflow!G187') # => outputs -2966.25862693
sp.set_value('InputData!G14', 0) # this is to avoid direct evaluation
sp.set_value('InputData!G14', 2025)
print 'Second evaluation', sp.evaluate('Cashflow!G187') # => outputs -3719.5504961
``` 

With `InputData!G14` as `2025` in the .XLS, 
````
print 'First evaluation', sp.evaluate('Cashflow!G187') # => outputs -2582.30664008
sp.set_value('InputData!G14', 0) # this is to avoid direct evaluation
sp.set_value('InputData!G14', 2025)
print 'Second evaluation', sp.evaluate('Cashflow!G187') # => outputs -2582.30663952
```

**Comment by vallettea** [comment]
?!
since I assume there is no random in your code, I guess it is because we compute subbranches in different orders and the calculus are non-commutatives ?

**Comment by iOiurson** [comment]
I really don't know... but there shouldn't be random in my code.
Have we already implemented the branch factorization into constants ? Maybe the input cell i modify is a part of a "constant" branch...

**Comment by vallettea** [comment]
i'm on it.

**Comment by vallettea** [comment]
but still we shouldn't rely on this

**Comment by iOiurson** [comment]
This was due to the fact that cells inside a range (but not at extremities) were not updated. This should be solved by #26. To be verified.

**Comment by iOiurson** [comment]
Closed by the different code evolutions on Range updates.
